packages/wdio-browserstack-service/src/accessibility-handler.ts

```diff
@@ -0,0 +1,343 @@
+        }
+
+        const pickleData = world.pickle
+        const gherkinDocument = world.gherkinDocument
+        const featureData = gherkinDocument.feature
+        const uniqueId = getUniqueIdentifierForCucumber(world)
+
+        if (isBrowserstackSession(this._browser)) {
+            const shouldScanScenario = shouldScanTestForAccessibility(featureData?.name, pickleData.name, this._accessibilityOptions)
+            this._testMetadata[uniqueId] = {
+                scanTestForAccessibility : shouldScanScenario,
+                accessibilityScanStarted : true
+            }
+
+            if (this._browser && isAccessibilityAutomationSession(this._accessibility)) {
+                let pageOpen = true
+
+                try {
+                    const currentURL = await (this._browser as WebdriverIO.Browser).getUrl()
+                    const url = new URL(currentURL)
+                    pageOpen = url?.protocol === 'http:' || url?.protocol === 'https:'
+                } catch (e) {
+                    pageOpen = false
+                }
+
+                try {
+                    if (pageOpen) {
+                        if (shouldScanScenario) {
+                            log.info('Setup for Accessibility testing has started. Automate test case execution will begin momentarily.')
+                            await (this._browser as WebdriverIO.Browser).executeAsync(`
+                                const callback = arguments[arguments.length - 1]
+                                const fn = () => {
+                                    window.addEventListener('A11Y_TAP_STARTED', fn2)
+                                    const e = new CustomEvent('A11Y_FORCE_START')
+                                    window.dispatchEvent(e)
+                                }
+                                const fn2 = () => {
+                                    window.removeEventListener('A11Y_TAP_STARTED', fn)
+                                    callback()
+                                }
+                                fn()
+                            `)
+                        } else {
+                            await (this._browser as WebdriverIO.Browser).execute(`
+                                const e = new CustomEvent('A11Y_FORCE_STOP')
+                                window.dispatchEvent(e)
+                            `)
+                        }
+                        this._testMetadata[uniqueId].accessibilityScanStarted = shouldScanScenario
+
+                        if (shouldScanScenario) {
+                            log.info('Automate test case execution has started.')
+                        }
+                    }
+                } catch (error) {
+                    log.error(`Exception in starting accessibility automation scan for this test case ${error}`)
+                }
+            }
+        }
+    }
+
+    async afterScenario (world: ITestCaseHookParameter) {
+        if (!this._accessibility) {
```

Comment:
At this point, maybe have a function that determines whether a hook should be run or not.
```suggestion
        if (!this._accessibility || !isBrowserstackSession(this._browser)) {
```